blockstore/splitstore/splitstore_compact.go

```diff
@@ -631,14 +631,16 @@ func (s *SplitStore) endTxnProtect() {
 	s.txnMissing = nil
 }
 
-func (s *SplitStore) walkChain(ts *types.TipSet, inclState abi.ChainEpoch, inclMsgs abi.ChainEpoch,
+func (s *SplitStore) walkChain(ts *types.TipSet, inclState, inclMsgs, inclMsgReceipts abi.ChainEpoch,
```

Comment:
Why separate height for receipts?

Reply:
well, I wanted to decouple it from state as it was the source of the bug -- when we sync from snapshot we dont have message receipts at all!

We could ostensibly couple them with the messages, but I don't think we need them at all in order to support the use cases -- please correct me if I am wrong :)

Reply:
following sync discussion, we discussed to collate them with message retention.